### PR TITLE
Feature/tinyobj update polishing

### PR DESCRIPTION
**Pull request by jefferis**

### CODE REVIEW

File: R/loadobj.R [hunk 1]
```diff
@@ -9,16 +9,20 @@
...
 #'   structures after the first code snapshot was taken for the this package in
-#'   2015. In order to benefit from bug fixes, we have updated the coded in
-#'   2020, but we note that \code{tinyobjloader} now de-duplicates vertices more
-#'   aggressively e.g. in the situation where there are texture coordinates.
+#'   2015. In order to benefit from bug fixes, we updated the code in 2020 but
+#'   we note that \code{tinyobjloader} now de-duplicates vertices more
+#'   aggressively e.g. in the situation where there are texture coordinates. We
+#'   were forced to revert these de-duplication on the R side in order to allow
+#'   texture coordinate mapping for display in rgl; note that this only happens
+#'   when there are texture coordinates or normals in the obj file.
```
Comment (jefferis) [inline R/loadobj.R:25]: @trevorld correct?

File: tests/testthat/test-loadobj.R [hunk 2]
```diff
@@ -53,3 +53,16 @@ test_that("can convert to rgl format", {
...
+
+test_that("can load more complex obj files", {
+  expect_is(tile<-read.obj(system.file("obj/tile.wavefront", package = "readobj"), triangulate = FALSE, convert.rgl = TRUE),
+            "shapelist3d")
+
+    expect_is(pawn <-
+              read.obj(system.file("obj/pawn.wavefront", package = "readobj")),
+            "list")
+  # try without triangulating
+  expect_error(read.obj(system.file("obj/pawn.wavefront", package = "readobj"), triangulate = FALSE))
```
Comment (jefferis) [inline tests/testthat/test-loadobj.R:66]: @trevorld note this change in behaviour. But at least pawn is now read correctly when triangulate=TRUE

File: R/loadobj.R [hunk 3]
```diff
@@ -9,16 +9,20 @@
 #'   \code{rgl::\link[rgl]{mesh3d}} objects.
 #' @param triangulate (default \code{TRUE}) Whether to convert all mesh faces to
 #'   triangles. Note that only meshes with triangular or quad faces are
-#'   supported.
+#'   supported, so setting \code{triangulate=FALSE} will throw an error for more
+#'   complex files.
```
Comment (jefferis) [inline R/loadobj.R:13]: @trevorld ok?

**Comment by jefferis** [comment]
@trevorld note that I have added your `pawn` file as a complex test as previously discussed. I think this is now a candidate for CRAN, but I need to upgrade to R 4.1.0 and do revdepchecks before submitting.

File: R/loadobj.R (hunk 3 above)
Comment (trevorld) [reply R/loadobj.R:13]: Looks good to me.

File: R/loadobj.R (hunk 1 above)
Comment (trevorld) [reply R/loadobj.R]: I commited to this PR a tweak

**Review by trevorld — APPROVED** [review]
* Looks good
* I added a commit which made minor tweaks to one of the man Details
* Technically the modified returned data structure (when not converting to ``rgl``) may be considered a `BREAKING CHANGE`.  You have already highlighted the relevant changes in the NEWS but perhaps worth emphasizing a bit more in a separate "Breaking Changes" section in NEWS?

**Pull request merged**